src/api.py

```diff
@@ -0,0 +1,123 @@
+    engine = p.create_engine()
+    columns=['order_id', 'seller_id', 'customer_unique_id', 'product_id', 'price', 'freight_value']
+    data = p.load_data_from_db('orders', engine, columns)
+    result = p.get_sales_analysis(data)
+    return jsonify(json.loads(result.write_json()))
+
+@app.route('/api/update_analytics', methods=['POST'])
+def update_analytics():
+    """Update analytics tables in the database"""
+    engine = p.create_engine()
+    p.analyze_and_load('orders', engine)
+    return jsonify({'message': 'Analytics tables updated successfully'})
+
+@app.route('/api/orders_by_date', methods=['GET'])
+def orders_by_date():
+    """Get orders within a specific date range (yyyy-mm-dd)"""
+    start_date = request.args.get('start_date')
+    end_date = request.args.get('end_date')
+    engine = p.create_engine()
+    columns=['order_id', 'order_purchase_timestamp', 'order_delivered_customer_date']
+    orders_df = p.load_data_from_db('orders', engine, columns)
+    result = p.get_orders_by_date(orders_df, start_date, end_date)
+    return jsonify(json.loads(result.write_json()))
+
+@app.route('/api/top_customers', methods=['GET'])
+def top_customers():
+    """Get top N customers by total spending"""
+    n = int(request.args.get('n', 10))
+    engine = p.create_engine()
+    columns=['customer_unique_id', 'price', 'item_quantity']
+    orders_df = p.load_data_from_db('orders', engine, columns)
+    result = p.get_top_customers(orders_df, n)
+    return jsonify(json.loads(result.write_json()))
+
+@app.route('/api/orders_by_customer', methods=['GET'])
+def orders_by_customer():
+    """Get orders for a specific customer"""
+    customer_id = request.args.get('customer_id')
+    engine = p.create_engine()
+    columns=['order_id', 'customer_unique_id', 'price']
+    orders_df = p.load_data_from_db('orders', engine, columns)
+    result = p.get_orders_by_customer(orders_df, customer_id)
+    return jsonify(json.loads(result.write_json()))
+
+@app.route('/api/orders_by_seller', methods=['GET'])
+def orders_by_seller():
+    """Get orders for a specific seller"""
+    seller_id = request.args.get('seller_id')
+    engine = p.create_engine()
+    columns=['order_id', 'seller_id', 'price']
+    orders_df = p.load_data_from_db('orders', engine, columns)
+    result = p.get_orders_by_seller(orders_df, seller_id)
+    return jsonify(json.loads(result.write_json()))
+
+@app.route('/api/orders_by_product', methods=['GET'])
+def orders_by_product():
+    """Get orders for a specific product"""
+    product_id = request.args.get('product_id')
+    engine = p.create_engine()
+    columns=['order_id', 'product_id', 'product_category_name','product_category_name_english']
+    orders_df = p.load_data_from_db('orders', engine, columns)
+    result = p.get_orders_by_product(orders_df, product_id)
+    return jsonify(json.loads(result.write_json()))
+
+if __name__ == '__main__':
+    app.run(host='0.0.0.0', port=5000, debug=True)
```

Comment:
I'll let this pass now but a better practice to put things like this in environment variables, port and host.


Please ensure debug is False in prod, it's not a secure practice to turn debug to true in prod, more reasons you should use environment variables to control things like this.

Reply:
Ok. I've set debug to False.